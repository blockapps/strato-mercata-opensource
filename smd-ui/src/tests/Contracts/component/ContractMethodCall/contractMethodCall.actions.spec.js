import {
  methodCallOpenModal,
  methodCallCloseModal,
  methodCallFetchArgs,
  methodCallFetchArgsSuccess,
  methodCallFetchArgsFailure,
  methodCall,
  methodCallSuccess,
  methodCallFailure
} from '../../../../components/Contracts/components/ContractMethodCall/contractMethodCall.actions';
import { modals } from './contractMethodCallMock';

describe('ContractMethodCall: action', () => {


  describe('method call', () => {

    test('request', () => {
      expect(methodCall(modals.key, modals.payload)).toMatchSnapshot();
    });

    test('success', () => {
      expect(methodCallSuccess(modals.key, modals.result)).toMatchSnapshot();
    });

    test('failure', () => {
      const result = 'ERROR';
      expect(methodCallFailure(modals.key, result)).toMatchSnapshot();
    });

  })

});