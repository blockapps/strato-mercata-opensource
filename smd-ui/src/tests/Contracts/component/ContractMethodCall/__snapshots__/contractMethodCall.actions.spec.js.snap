// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractMethodCall: action method call failure 1`] = `
Object {
  "key": "methodCallgreet8070db2390462e2b5748085bde1350590e08bb17",
  "result": "ERROR",
  "type": "METHOD_CALL_FAILURE",
}
`;

exports[`ContractMethodCall: action method call request 1`] = `
Object {
  "key": "methodCallgreet8070db2390462e2b5748085bde1350590e08bb17",
  "payload": Object {
    "args": Object {},
    "contractAddress": "8070db2390462e2b5748085bde1350590e08bb17",
    "contractName": "Greeter",
    "methodName": "greet",
    "password": "pass",
    "userAddress": "76a3192ce9aa0531fe7e0e3489a469018c0bff03",
    "username": "tanuj",
  },
  "type": "METHOD_CALL_REQUEST",
}
`;

exports[`ContractMethodCall: action method call success 1`] = `
Object {
  "key": "methodCallgreet8070db2390462e2b5748085bde1350590e08bb17",
  "result": Array [
    Object {
      "data": Object {
        "contents": Array [
          "200",
        ],
        "tag": "Call",
      },
      "hash": "31bcbc86cb6fdd273443406ae27906485fccc8475b269de04878d9b797c0f47e",
      "status": "Success",
      "txResult": Object {
        "appName": "",
        "blockHash": "36d2629fb5ef27457eb70a87cd886284d0ac7c197f7b3eaf1102edbb8e3518b1",
        "chainId": null,
        "contractsCreated": "",
        "contractsDeleted": "",
        "deletedStorage": "",
        "etherUsed": "0000000000000000000000000000000000000000000000000000000005f5e100",
        "gasUsed": "0000000000000000000000000000000000000000000000000000000005f5e100",
        "kind": "SolidVM",
        "message": "Success!",
        "newStorage": "",
        "orgName": "Blockapps",
        "response": "(200)",
        "stateDiff": "",
        "status": "success",
        "time": 0.001361853,
        "trace": "",
        "transactionHash": "31bcbc86cb6fdd273443406ae27906485fccc8475b269de04878d9b797c0f47e",
      },
    },
  ],
  "type": "METHOD_CALL_SUCCESS",
}
`;
